### PR TITLE
Refactoring appearance logic out of RMVC and into Views

### PR DESCRIPTION
**Pull request by JThramer**
This PR concerns #994, where it is desired to refactor view-appearance logic out of the RMVC and into the appropriate views themselves. This makes things a bit more lonely coupled and easier to work with in a piecemeal fashion.

To-Do: 
- [x] InstructionsBannerView
- [x] NextBannerView
- [x] ~BottomBannerView?~ (already pretty well-encapsulated)

Phase 2 To-Do:
- [x] Create TopBannerViewController
- [ ] Plug it in to RMVC
- [ ] Investigate "seconds" ambiguity
- [ ] Make sure you remove all shims (such as `tooFar`)

/cc @bsudekum @1ec5 @frederoni @ericrwolfe

### CODE REVIEW

**Comment by bsudekum** [comment]
@JThramer feel free to break your todo list out into 3 different PRs.

**Comment by JThramer** [comment]
Turned out to be unnecessary @bsudekum. I wasn't able to make as much of an impact on `NextBannerView` as I would have liked, due to there being a bunch of logic rules around whether-or-not the Next banner should show (are the "lanes" showing? etc)... We may want to consider breaking everything in the "top" out into it's own stack-view that could make those determinations itself, but that seems like a big lift for this PR, especially considering #981 is still in-flight.

File: MapboxNavigation/InstructionsBannerView.swift [hunk 1]
```diff
@@ -71,6 +71,17 @@ open class BaseInstructionsBannerView: UIControl {
         delegate?.didTapInstructionsBanner(self)
     }
     
+    func update(progress: RouteLegProgress, location: CLLocation, secondsRemaining: TimeInterval) {
```
Comment (1ec5) [inline MapboxNavigation/InstructionsBannerView.swift]: `location` and `secondsRemaining` are unused, so  `setRouteProgress(_:)` would suffice as the method name.

File: MapboxNavigation/NextBannerView.swift [hunk 2]
```diff
@@ -70,4 +71,9 @@ open class NextBannerView: UIView {
         instructionLabel.rightAnchor.constraint(equalTo: rightAnchor, constant: -16).isActive = true
     }
     
+    func update(step: RouteStep, instruction: VisualInstruction) {
+        maneuverView.step = step
+        instructionLabel.instruction = instruction.primaryTextComponents
```
Comment (1ec5) [inline MapboxNavigation/NextBannerView.swift]: An `update(list:of:properties:to:update:)` method is still a pretty raw wrapper on the previous approach of setting all the properties directly and individually. This is an opportunity for us to think about when RouteMapViewController updates these properties and why. That’ll naturally lead us to a more descriptive name for the method, so we don’t inadvertently call or neglect to call it in the future.

In this case, I noticed that InstructionsBannerView has separate methods for updating the step (as part of setting the route progress) and setting the instruction; why not follow the same pattern for NextBannerView?

Comment (bsudekum) [reply MapboxNavigation/NextBannerView.swift]: Agreed, noting that `maneuverView.step ` is used for deciding upon the which maneuver arrow to draw.

Comment (bsudekum) [reply MapboxNavigation/NextBannerView.swift]: Ideally, a developer could just give us a RouteProgress, and we'd be able to draw the correct UI for them.

Comment (1ec5) [reply MapboxNavigation/NextBannerView.swift]: Good point. How about making as many of these classes as possible maintain a single RouteProgress-typed property? Each setter would take care of extracting the relevant information from that RouteProgress object. We could create a new protocol for classes that track state via a RouteProgress-typed property.

**Comment by JThramer** [comment]
Status update: Yesterday it was agreed among members of @mapbox/navigation-ios that the views associated with the "Top banner" should be extracted into their own containing class to take this concept even further. I am doing this by creating a `TopBannerViewController` and associated XIB.

File: MapboxNavigation/TopBannerViewController.swift [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    // MARK: - Model Updating
+    func update(progress: RouteProgress, location: CLLocation, secondsRemaining: TimeInterval) {
+        //update banner
+        instructions.update(progress: progress.currentLegProgress, location: location, secondsRemaining: secondsRemaining)
+        
+        //update lanes
+        if let upComingStep = progress.currentLegProgress?.upComingStep, !progress.currentLegProgress.userHasArrivedAtWaypoint {
+            updateLaneViews(step: upComingStep, durationRemaining: progress.currentLegProgress.currentStepProgress.durationRemaining)
```
Comment (JThramer) [inline MapboxNavigation/TopBannerViewController.swift]: Investigate why one view uses the `secondsRemaining` argument and another view seems to pull the same value(?) out of `RouteProgress`.

**Comment by JThramer** [comment]
This PR has been blocked by a major architectural change (#981) introduced upstream. I have started work on resolving this blocker, this effort is being tracked by #1062.

**Comment by JThramer** [comment]
This is ultra-stale, and good progress was made in-parallel with the issue that this PR was meant to address. Going to close it.

**Pull request closed without merging**